flink-connectors/flink-connector-kafka-base/src/test/java/org/apache/flink/streaming/connectors/kafka/KafkaProducerTestBase.java

```diff
@@ -243,10 +243,14 @@ protected void testOneToOneAtLeastOnce(boolean regularSink) throws Exception {
 		properties.putAll(secureProps);
 		// decrease timeout and block time from 60s down to 10s - this is how long KafkaProducer will try send pending (not flushed) data on close()
 		properties.setProperty("timeout.ms", "10000");
+		// KafkaProducer prior to KIP-91 (release 2.1) uses request timeout to expire the unsent records.
+		properties.setProperty("request.timeout.ms", "3000");
+		// KafkaProducer in 2.1.0 and above uses delivery timeout to expire the the records.
+		properties.setProperty("delivery.timeout.ms", "5000");
 		properties.setProperty("max.block.ms", "10000");
 		// increase batch.size and linger.ms - this tells KafkaProducer to batch produced events instead of flushing them immediately
 		properties.setProperty("batch.size", "10240000");
-		properties.setProperty("linger.ms", "10000");
+		properties.setProperty("linger.ms", "1000");
```

Comment:
Decreasing this from 10s to 1s might decrease the chance for the test do detect committing/flushing failures. However 1s should still be enough on at least some of the travis builds to detect it.

Also, are those config adjustments related to " Fix the cascading failure in KafkaProducerTestBase"? If some of them are about speeding up the test, it should be a separate commit.

Reply:
We can leave this at 10 seconds. I don't think this would matter much.

Reply:
We can probably keep this value to 10 seconds then. And yes, your are right, we should have a separate commit for accelerating the tests.